sionna/channel/optical/fiber.py

```diff
@@ -0,0 +1,309 @@
+#
+# SPDX-FileCopyrightText: Copyright (c) 2021-2022 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+#
+# Author: Tim Alexander Uhlemann <uhlemann@ieee.org>
+
+"""
+This module defines the split-step Fourier method to approximate the solution of
+the nonlinear Schroedinger equation.
+"""
+
+import tensorflow as tf
+from tensorflow.keras.layers import Layer
+import sionna
+from sionna.channel import utils
+
+
+class SSFM(Layer):
+    # pylint: disable=line-too-long
+    r"""SSFM(alpha, beta_2, f_c, gamma, half_window_length, ell, n_ssfm, n_sp, dt, t_norm, with_amplification, with_attenuation, with_dispersion, with_nonlinearity, swap_memory=True, dtype=tf.complex64, **kwargs)
+
+    Layer implementing the split-step Fourier method (SSFM).
+
+    The SSFM (first mentioned in [SSFM]_) is implemented as
+    symmetrized SSFM according to Eq. (7) of [SymSSFM]_ and can be written as
+
+    .. math::
+
+        E(z+\Delta_z,t) \approx \exp\left(\frac{\Delta_z}{2}\hat{D}\right)\exp\left(\int^{z+\Delta_z}_z \hat{N}(z')dz'\right)\exp\left(\frac{\Delta_z}{2}\hat{D}\right)E(z,\,t)
+
+    The integral is approximated by :math:`\Delta_z\hat{N}`.
+
+    Additionally, ideally distributed Raman amplification may be applied, which
+    is implemented as in [RamanASE]_.
+
+    The SSFM operates on normalized time :math:`T_\text{norm}` (e.g., 1 ps) and
+    distance units :math:`L_\text{norm}` (e.g., 1 km). Hence, parameters as well
+    as the signal itself have to be given in the same normalized units. Note
+    that, despite the normalization, the SSFM is implemented with physical units
+    different from the normalization, e.g., used for the nonlinear
+    Fourier transform.
+
+    This class inherits from the Keras `Layer` class and can be used as layer in
+    a Keras model.
+
+    Example
+    --------
+
+    Setting-up:
+
+    >>> ssfm_channel = SSFM(
+    >>>     0.046, -21.67, 193.55e12, 1.27, 100, 80, 200, 1e-17, 1.0,
+    >>>     True, True, True, True)
+
+    Running:
+
+    >>> y = ssfm_channel([1.0+1.0j, 1.0+1.0j, 1.0+1.0j])
+
+    Parameters
+    ----------
+        alpha : float
+            Attenuation coefficient :math:`\alpha` in :math:`(1/L_\text{norm})`
+
+        beta_2 : float
+            Chromatic dispersion coefficient :math:`\beta_2` in :math:`(T_\text{norm}^2/L_\text{norm})`
+
+        f_c : float
+            Carrier frequency :math:`f_\mathrm{c}` in :math:`(1/T_\text{norm})`
+
+        gamma : float
+            Kerr-nonlinearity :math:`\gamma` in :math:`(1/L_\text{norm}/W)`
+
+        half_window_length : int
+            Half of the Hammwing window length
+
+        ell : float
```

Comment:
I suggest to rename "ell" as this is just the name of the symbol in latex. How about length?

Reply:
OK